### DISCUSSION TITLE
pull access denied when accessing ECR repositories

### DISCUSSION THREAD

**Discussion by rainkinz** [abiosoft/colima#647] [Q&A] [answered]
Hi, I am just starting out with colima and works great for one of my projects, however for a project that uses a private ECR repository I am getting a `pull access denied` error. I am making sure to login using:

```
aws --profile=PROFILE ecr get-login-password | docker login --username AWS --password-stdin ACCOUNT_NUMBER.dkr.ecr.us-east-1.amazonaws.com
```

My `~/.docker/config.json` looks like this:

```
{
	"auths": {
		"ACCOUNT_NUMBER.dkr.ecr.us-east-1.amazonaws.com": {}
	},
	"credsStore": "osxkeychain",
	"currentContext": "colima"
}
```

FYI I tried the setting with and without the `credStore` entry 

Thanks

**Comment by abiosoft** [comment, marked as answer]
I think you should use ecr-credential-helper https://github.com/awslabs/amazon-ecr-credential-helper and modify your docker config to something to include the following.

```json
{
  "credHelpers": {
    "public.ecr.aws": "ecr-login",
    "<AWS_ACCOUNT_ID>.dkr.ecr.<REGION>.amazonaws.com": "ecr-login"
  }
}
```

**Reply by rainkinz** [reply]
Thank you, that seems to have done the trick.
